app/code/Magento/MediaGalleryRenditions/Plugin/RemoveRenditions.php

```diff
@@ -0,0 +1,84 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+declare(strict_types=1);
+
+namespace Magento\MediaGalleryRenditions\Plugin;
+
+use Magento\Framework\App\Filesystem\DirectoryList;
+use Magento\Framework\Filesystem;
+use Magento\MediaGalleryApi\Api\DeleteAssetsByPathsInterface;
+use Magento\MediaGalleryRenditionsApi\Api\GetRenditionPathInterface;
+use Psr\Log\LoggerInterface;
+
+/**
+ * Remove renditions when assets are removed
+ */
+class RemoveRenditions
+{
+    /**
+     * @var GetRenditionPathInterface
+     */
+    private $getRenditionPath;
+
+    /**
+     * @var Filesystem
+     */
+    private $filesystem;
+
+    /**
+     * @var LoggerInterface
+     */
+    private $log;
+
+    /**
+     * @param GetRenditionPathInterface $getRenditionPath
+     * @param Filesystem $filesystem
+     * @param LoggerInterface $log
+     */
+    public function __construct(
+        GetRenditionPathInterface $getRenditionPath,
+        Filesystem $filesystem,
+        LoggerInterface $log
+    ) {
+        $this->getRenditionPath = $getRenditionPath;
+        $this->filesystem = $filesystem;
+        $this->log = $log;
+    }
+
+    /**
+     * Remove renditions when assets are removed
+     *
+     * @param DeleteAssetsByPathsInterface $deleteAssetsByPaths
+     * @param \Closure $proceed
+     * @param array $paths
+     */
+    public function aroundExecute(
```

Comment:
It would be much better to use `after` plugin instead of `around`

Reply:
After plugin does not have access to parameters

Reply:
Please, check the document: https://devdocs.magento.com/guides/v2.4/extension-dev-guide/plugins.html Parameters in after plugins are already supported.

Reply:
Thanks, will try this out